src/components/Dialogs/PreferenceDialog/PreferenceDialogComponent.tsx

```diff
@@ -55,19 +55,24 @@ export class PreferenceDialogComponent extends React.Component {
     };
 
     @computed get pvPreviewCubeSizeMaxValue(): number {
-        if (PreferenceStore.Instance.pvPreivewCubeSizeLimitUnit === MemoryUnit.TB) {
+        if (this.showedPvPreviewCubeSizeLimitUnit === MemoryUnit.TB) {
             return PV_PREVIEW_CUBE_SIZE_LIMIT / 1e12;
-        } else if (PreferenceStore.Instance.pvPreivewCubeSizeLimitUnit === MemoryUnit.GB) {
+        } else if (this.showedPvPreviewCubeSizeLimitUnit === MemoryUnit.GB) {
             return PV_PREVIEW_CUBE_SIZE_LIMIT / 1e9;
-        } else if (PreferenceStore.Instance.pvPreivewCubeSizeLimitUnit === MemoryUnit.MB) {
+        } else if (this.showedPvPreviewCubeSizeLimitUnit === MemoryUnit.MB) {
             return PV_PREVIEW_CUBE_SIZE_LIMIT / 1e6;
-        } else if (PreferenceStore.Instance.pvPreivewCubeSizeLimitUnit === MemoryUnit.kB) {
+        } else if (this.showedPvPreviewCubeSizeLimitUnit === MemoryUnit.kB) {
             return PV_PREVIEW_CUBE_SIZE_LIMIT / 1e3;
         } else {
             return PV_PREVIEW_CUBE_SIZE_LIMIT;
         }
     }
 
+    @computed get showedPvPreviewCubeSizeLimit(): number {
+        const preference = PreferenceStore.Instance;
+        return this.showedPvPreviewCubeSizeLimitUnit === MemoryUnit.GB ? preference.pvPreviewCubeSizeLimit : preference.pvPreviewCubeSizeLimit * 1000;
```

Comment:
I recommend normalizing the preferences in the upgrade function: if the unit pref exists, if it isn't GB, convert the limit value to GB and save, and also clear the unit pref. Don't save or use the unit pref anywhere else.

We can't remove it from the preferences store yet, because `clearPreferences` expects valid `PreferenceKeys`, so I suggest that we just mark it as deprecated in the schema and stop using it, and only remove it fully in a few releases' time.